apps/06.1Checkers.js

```diff
@@ -47,21 +53,91 @@ function Board() {
       // join the rowOfCheckers array to a string, separated by a space
       string += rowOfCheckers.join(' ');
       // add a 'new line'
-      string += "\n";
-    }
+      string += "\n"; 
+      }
     console.log(string);
   };
-
   // Your code here
-}
-function Game() {
+  this.checkers = [];
+  this.createCheckers = function() {
+    //define the white and black position array as given in the specs
+    var whitePositions = [
+      [0, 1], [0, 3], [0, 5], [0, 7], 
+      [1, 0], [1, 2], [1, 4], [1, 6], 
+      [2, 1], [2, 3], [2, 5], [2, 7],
+    ];
+    var blackPositions = [
+      [5, 0], [5, 2], [5, 4], [5, 6], 
+      [6, 1], [6, 3], [6, 5], [6, 7], 
+      [7, 0], [7, 2], [7, 4], [7, 6],
+    ];
+    // a better way to loop through using for...of... loop
+    //for (var spot of whiteCheckers) {
+    //   var row = spot[0];
+    //   var column = spot[1];
+    //   var whiteChecker = new Checker('white');
+    //   this.grid[row][column] = whiteChecker;
+    // }
+    for (var i = 0; i <= 11; i++) {
+      //instantiate a white checker
+      var whiteChecker = new Checker('white');
+      //place the checker on the grid at the corresponding index 
+      this.grid[whitePositions[i][0]][whitePositions[i][1]] = whiteChecker;
+      //push the checkers into the checker array 
+      this.checkers.push(whiteChecker);
+      //do the same thing for the black checker 
+      var blackChecker = new Checker('black');
+      this.grid[blackPositions[i][0]][blackPositions[i][1]] = blackChecker;
+      this.checkers.push(blackChecker);
+    }
+  }
+  //help function for selecting checkers on board
+  this.selectChecker = function(checkerPosition) {
+    //get the row and column indexes of the position passed in 
+    var row = checkerPosition[0];
+    var column = checkerPosition[1];
+    //put the checker in the variable c 
+    var c = this.grid[row][column];
+    //nullify the checker on board
+    this.grid[row][column] = null;
+    return c;
+  }
+  //helper function for killing a checker 
+  this.killChecker = function(position) {
+    //get the checker using the helper function 
+    var checkerForKill = this.selectChecker(position);
+    //get the index of the checker to be killed 
+    var killPosition = this.checkers.indexOf(checkerForKill);
+    //remove the checker from the array 
+    this.checkers.splice(killPosition,1);
+    //assign the checker's position on the grid to null 
+    this.grid[position[0]][position[1]] = null;
+    }
+};
 
+function Game() {
   this.board = new Board();
-
   this.start = function() {
     this.board.createGrid();
-    // Your code here
+    // Call the createCheckers() function and map out all the checkers on the grid 
+    this.board.createCheckers();
   };
+  this.moveChecker = (start, end) => {
+    //select the checker and put it into the 'checker' variable 
+    var checker = this.board.selectChecker(start);
```

Comment:
If you're going to add additional functionality to the `selectChecker()` method, I would recommend changing the method name. 

Ex: `moveChecker`